### ISSUE TITLE
Failure in test_warn_non_seekable_files

### ISSUE THREAD

**Issue by jakirkham** [dask/dask#2372]
```python
________________________ test_warn_non_seekable_files _________________________

capsys = <_pytest.capture.CaptureFixture instance at 0x7f888ab837e8>

    def test_warn_non_seekable_files(capsys):
        files2 = valmap(compress['gzip'], csv_files)
        with filetexts(files2, mode='b'):
    
            # with tm.assert_produces_warning(UserWarning):
            df = dd.read_csv('2014-01-*.csv', compression='gzip')
            assert df.npartitions == 3
            out, err = capsys.readouterr()
>           assert 'gzip' in err
E           AssertionError: assert 'gzip' in ''
dask/dataframe/io/tests/test_csv.py:433: AssertionError
```

ref: https://travis-ci.org/dask/dask/jobs/235413731#L3890

**Comment by jakirkham** [comment]
This is also showing up on `master` where it wasn't previously.

ref: https://travis-ci.org/dask/dask/jobs/235211927#L3882

**Comment by jcrist** [comment]
Fixed by #2370.
